lightning/src/ln/peer_handler.rs

```diff
@@ -1524,10 +1532,12 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 
 				for (_, peer_mutex) in peers.iter() {
 					let mut peer = peer_mutex.lock().unwrap();
-					if !peer.channel_encryptor.is_ready_for_encryption() || peer.their_features.is_none() ||
+					if peer.their_features.is_none() ||
```

Comment:
Is this check redundant with `!peer.should_forward_channel_announcement(msg.contents.short_channel_id)`?

Similarly for the checks for the other messages.